wolfcrypt/src/asn.c

```diff
@@ -9117,64 +9117,54 @@ int wc_EccPublicKeyDecode(const byte* input, word32* inOutIdx,
                           ecc_key* key, word32 inSz)
 {
     int    length;
-    int    ret = 0;
+    byte   b;
 
     if (input == NULL || inOutIdx == NULL || key == NULL || inSz == 0)
         return BAD_FUNC_ARG;
 
     if (GetSequence(input, inOutIdx, &length, inSz) < 0)
         return ASN_PARSE_E;
 
-#if defined(OPENSSL_EXTRA) || defined(ECC_DECODE_EXTRA)
-    {
-        byte b = input[*inOutIdx];
-        if (b != ASN_INTEGER) {
-            /* not from decoded cert, will have algo id, skip past */
-            if (GetSequence(input, inOutIdx, &length, inSz) < 0)
-                return ASN_PARSE_E;
+    if (GetSequence(input, inOutIdx, &length, inSz) < 0)
```

Comment:
Please resolve conflicts to easily test your changes as both code paths WERE working exactly as intended.

Both code paths did work.  By default ECC_DECODE_EXTRA was on unless the user turned on WOLFSSL_NO_DECODE_EXTRA.  'make check' loads ecc public keys from certs during cipher suite tests and works whether WOLFSSL_NO_DECODE_EXTRA is defined.  This is easy to test.

To force the OpenSSL specific encoding define WOLFSSL_CERT_EXT.  This tries to decode ecc-client-keyPub.  This works unless WOLFSSL_NO_DECODE_EXTRA is defined (which is intended).  Some people want to save code space and know they'll never be loading OpenSSL specific ecc public keys explicitly.

Just looking at the changes makes me think this no longer works as intended.  It can't be easily tested with the merge conflict in this exact area.